src/gpio/gpio.c

```diff
@@ -61,68 +62,107 @@ mraa_gpio_init_internal(mraa_adv_func_t* func_table, int pin)
         return NULL;
 
     mraa_result_t status = MRAA_SUCCESS;
-    char bu[MAX_SIZE];
-    int length;
+    mraa_board_t* board = plat;
+    struct stat dir_stat;
+    unsigned int phy_pin;
+    char bu[MAX_SIZE], pin_name[MRAA_PIN_NAME_SIZE];
+    int export_fd;
+    int length, i;
 
     mraa_gpio_context dev = (mraa_gpio_context) calloc(1, sizeof(struct _gpio));
     if (dev == NULL) {
         syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context", pin);
         return NULL;
     }
-
     dev->advance_func = func_table;
     dev->pin = pin;
-
+    dev->gpio_path = (char*) calloc(MAX_SIZE, sizeof(char));
+    if (dev->gpio_path == NULL) {
+        syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context's attribute", pin);
+        status = MRAA_ERROR_INVALID_RESOURCE;
+        goto init_internal_cleanup;
+    }
     if (IS_FUNC_DEFINED(dev, gpio_init_internal_replace)) {
         status = dev->advance_func->gpio_init_internal_replace(dev, pin);
         if (status == MRAA_SUCCESS)
             return dev;
         else
             goto init_internal_cleanup;
     }
-
```

Comment:
surplus styling fix (BTW doesn't it make it harder to read, by removing whitespace between logical blocks?)